frontend/components/panels/DesktopClassPanel.js

```diff
@@ -24,16 +25,32 @@ import globe from './globe.svg';
 
 
 export default class DesktopClassPanel extends BaseClassPanel {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.reRenderCheckBox = this.reRenderCheckBox.bind(this);
+  }
+
   static propTypes = {
     aClass: PropTypes.object.isRequired,
   };
 
+
   componentDidUpdate() {
     macros.debounceTooltipRebuild();
   }
 
   componentDidMount() {
     macros.debounceTooltipRebuild();
+    user.registerCallback(this.reRenderCheckBox);
+  }
+
+  componentWillUnmount() {
+    user.unregisterCallback(this.reRenderCheckBox);
+  }
+
+  reRenderCheckBox() {
+    this.setState({ test: !this.state.test });
```

Comment:
test ?

Reply:
it was to re render the checkbox... is there a more concrete way than this?